front/src/applications/operationalStudies/hooks/useScenarioData.ts

```diff
@@ -0,0 +1,207 @@
+      const projectedTrainsById = keyBy(projectedTrains, 'id');
+
+      _trainIdsToRemove.forEach((trainId) => {
+        delete trainSchedulesById[trainId];
+        delete trainScheduleSummariesById[trainId];
+        delete projectedTrainsById[trainId];
+      });
+
+      const newTrainSchedules = Object.values(trainSchedulesById);
+      if (trainSchedules && newTrainSchedules.length !== trainSchedules.length)
+        setTrainSchedules(newTrainSchedules);
+
+      if (Object.values(trainScheduleSummariesById).length !== trainScheduleSummaries.length)
+        setTrainScheduleSummariesById(trainScheduleSummariesById);
+
+      if (Object.keys(projectedTrainsById).length !== projectedTrains.length)
+        setProjectedTrainsById(projectedTrainsById);
+
+      setTrainIdsToRemove([]);
+    };
+
+    if (trainIdsToRemove.length > 0) removeTrains(trainIdsToRemove);
+  }, [trainIdsToRemove]);
+
+  // TODO: refacto selectTrainAndPathForProjection
+  useEffect(() => {
+    if (trainSchedules && infra?.state === 'CACHED' && trainSchedules.length > 0) {
+      selectTrainAndPathForProjection(
+        trainSchedules.map((trainSchedule) => trainSchedule.id),
+        setSelectedTrainId,
+        setTrainIdUsedForProjection,
+        trainIdUsedForProjection,
+        selectedTrainId
+      );
+    }
+  }, [trainSchedules, infra]);
+
+  const upsertTrainSchedules = useCallback(
+    (trainSchedulesToUpsert: TrainScheduleResult[]) => {
+      const newTrainSchedulesById = {
+        ...keyBy(trainSchedules, 'id'),
+        ...keyBy(trainSchedulesToUpsert, 'id'),
+      };
+      const newTrainSchedules = sortBy(Object.values(newTrainSchedulesById), 'start_time');
+      setTrainSchedules(newTrainSchedules);
+
+      const sortedTrainSchedulesToUpsert = sortBy(trainSchedulesToUpsert, 'start_time');
+      setTrainIdsToFetch(sortedTrainSchedulesToUpsert.map((trainSchedule) => trainSchedule.id));
+    },
+    [trainSchedules]
+  );
+
+  // TODO: remove the 5 following hooks when trainIdUsedForProjection and selectedTrainId
+  // are managed in the state
+  useEffect(() => {
+    if (selectedTrainId !== selectedTrainIdInStore)
+      dispatch(updateSelectedTrainId(selectedTrainId));
+  }, [selectedTrainId]);
+  useEffect(() => {
+    if (selectedTrainId !== selectedTrainIdInStore) setSelectedTrainId(selectedTrainIdInStore);
+  }, [selectedTrainIdInStore]);
+
+  useEffect(() => {
+    if (trainIdUsedForProjection !== trainIdUsedForProjectionInStore)
+      dispatch(updateTrainIdUsedForProjection(trainIdUsedForProjection));
+  }, [trainIdUsedForProjection]);
+  useEffect(() => {
+    if (trainIdUsedForProjection !== trainIdUsedForProjectionInStore)
+      setTrainIdUsedForProjection(trainIdUsedForProjectionInStore);
+  }, [trainIdUsedForProjectionInStore]);
+
+  useEffect(() => {
+    dispatch(updateTrainIdUsedForProjection(undefined));
+    dispatch(updateSelectedTrainId(undefined));
+  }, []);
```

Comment:
I agree, we should simplify, but the best way would be to keep the state version because I think there is no need to store this in the store (since everything is recomputed at each refresh).
But if I need to implement the state version in this PR, this will involve a lot of change :/ I wanted to do it in another PR :/

What do you prefer:
- no state at all for this PR, and we replace the store by the state in another PR
- ok for this version of the code, and we replace the store by the state in another PR (cohabitation of the store and state version for a moment)
- you don't agree and prefer the store, then we can discuss it tomorrow at the front meeting ?

Reply:
let's go with the second option. If everything is recalculated each time and the data isn't reused elsewhere or needs to be persisted, it makes sense to keep it in the local state. However, let's add a TODO and create an issue for this. We can link the issue to this discussion thread to keep track of it. Does that sound good?

Reply:
Sorry, I thought about it again and you're right. I have removed the state and keep the store version :+1: